src/Screens/All.tsx

```diff
@@ -1,20 +1,33 @@
 import React from "react";
 
 import { View, StyleSheet, Image, SafeAreaView } from "react-native";
-import PokemonList from "@components/PokemonList";
+import PokemonList, { useScrollHandler } from "@components/PokemonList";
 import usePokemons from "../hooks/usePokemons";
 import { TabScreenProps } from "./types";
+import { useAnimatedStyle, useSharedValue } from "react-native-reanimated";
+import debounce from "lodash/debounce";
+import { SEARCH_HEIGHT } from "../constants";
 
 type Props = TabScreenProps<"All">;
 
 const All = ({ navigation }: Props) => {
+  const [search, _setSearch] = React.useState("");
+  const setSearch = debounce(_setSearch, 500);
```

Comment:
🐛  - do you know why? 👀

Reply:
yep, it creates the new debounced function on each rerender. In this case, it might be fixed with useMemom because `_setSearch ` shouldn't change the reference. Typically beginner mistake 😳